### ISSUE TITLE
Doesn't works with two-factor authentication

### ISSUE THREAD

**Issue by tylon85** [nextcloud/notes#328]
I activated currently the integrated two-factor authentication in Nextcloud.

I have created an app password (without that I can't use the Notes app anymore) and now I got some edited notes twice (with 2 in the name of the note) in the android notes app.

If I try to open a note via web interface, I got the following error:
"Verschlüsselungsfehler(tylon85: decoding/coding error): (filename.txt)multikeydecrypt with share key failed:error:0407109F:rsa routines:RSA_padding_check_PKCS1_type_2:pkcs decoding error"
and I can't open the note.

**To Reproduce**
Activate the integrated two-factor authentication of Nextcloud. Then set up a app password for the notes app and try to open a note via the app and via the web interface.

**Server:**
- Notes app version: 0.24.3
- Nextcloud version: 16.0.1.1
- OS: Android
- Web server: Apache
- PHP version: 7.1.30
- Database: unknown

**Client (please complete the following information):**
 - Browser (incl. version): Firefox 67.0.3
 - OS: Windows and Linux (Fedora)

**Log files**
The Nextcloud service is not hosted by me. So I can't get a log file.

**Comment by fozters** [comment]
It works (atleast used to work now i have some java.lang issue) with creating another totp key for notes, with single totp it does not work. Create another totp key, look for already closed issues..

**Comment by tylon85** [comment]
I created already another totp key. But I can't still read/access the already created notes.

**Comment by korelstar** [comment]
I finally found some time to have a look at this. I've activated 2FA using the TOTP provider app. As expected, using the 3rd party API works only after creating an app password.

However, I had no problems using the API or the web app after activating 2FA. The listed exception did not occur on my instance.

I'm wondering if your exception may come from another problem. The exception text indicates that you are using server-side encryption. Is this true? Maybe this is related: https://github.com/nextcloud/server/issues/8349

Are you sure that the problem comes from 2FA? Please try to temporarily deactivate 2FA for your user and try again using the API.

**Comment by tylon85** [comment]
Yes, server-side encryption is activated and I can't deactivates it.

Even if I don't use the 2FA per TOTP provider app the issue is still there. But everything works fine with calendar, contacts and so on, but that is not really important.

If I lost my notes forever, that would be the worst.

**Comment by tylon85** [comment]
If I try to open a text file in the Notes directory, I got this error message: "Nextcloud - Service Unavailable"

If I try to download a text file I got this error message: "Can't read file - multikeydecrypt with share key failed:error:0407109F:rsa routines:RSA_padding_check_PKCS1_type_2:pkcs decoding error".

**Comment by korelstar** [comment]
I'm pretty sure, that the origin of this issue is not the 2FA but your encryption setup. Personally, I have not that much experience with server-side encryption or end-to-end encryption.

Your provider uses server-side encryption. But what do you mean by "user directory encryption"?

By the way: the app doesn't support end-to-end encryption! If you use that, please deactivate it!

However, the error message sounds still related to https://github.com/nextcloud/server/issues/8349

**Comment by tylon85** [comment]
"user directory encryption": It's called "Encrypt the home storage" in the Server-side encryption section.

**Comment by teatreeoilchocolate** [comment]
The update to the nextcloud-note-android-app solves this problem.
Update and it should work.
At least it did for me.

**Comment by korelstar** [comment]
It look like this issue has been solved resp. is not related to 2FA. If there is still a problem with the latest versions of Nextcloud server, Nextcloud Notes (server app) and Nextcloud Notes (Android app) please open a new issue.
